### DISCUSSION TITLE
URL Lock

### DISCUSSION THREAD

**Discussion by Gabenny** [SalesforceCommerceCloud/sfcc-ci#539] [Team Posts]
Hello,

I would to lock page rooting for the folowing scenario:

- An ISML Page contains an Single-page application.
- the ISML is called via a controller.
- The result page URL could be like this "..../Mydev/"

I would  like to make sure that even if the SPA changes the URL to for example "/Mydev/something/blabla" the user keeps on the same ISML Page without reloading it.

Can you please tell me if there is a native SFCC configuration that could handle this.

If not so can you suggest a solution?

Thank you.

Best regards.

**Comment by GeneralPatten** [comment]
In Business Manager, add a URL Redirect with Source Type URI "/Mydev*" and Destination Type "/MyDev".

**Comment by pravngaur** [comment]
Hi Amine,

I am sorry, what is SPA?

with the limited understanding of your problem, you can use url redirects
for all the new /changes urls.

**Comment by mdzafar** [comment]
Hi @Gabenny - If you are trying to build single page app on SFCC then I think as a reference you should look at the checkout implementation in SFRA. There also after landing on checkout page, URL "Checkout-Begin" remains same only content on the same page gets updated based on the stage of checkout user is currently in (like shipping, payment)

I think you will need to play around a lot with AJAX to implement this because much of the work will happen on the client side. 

Even in SFRA due to Ajax, checkout handling is a bit complex & lot of stuff is happening in client side JavaScript .

Please let me know if you need further help.

**Comment by Gabenny** [comment]
Hello @mdzafar ,
Thank you for the reply.
The SPA will be integrated inside an ISML Page witch is called using a controller (I'm talking about the ISML page)
The SPA will be used to customize a product.
After product customization it has to be added to the cart.  I think that it's possible for the SPA to add a product inside the cart using AJAX.
Since the ISML Page will be hosting the SPA I would like to know first if the SPA changes the URL like for example "MyDev/Customization/sockets/Red" I want to make sure that I keep serving the same ISML page without reloading it and  and ignore the part "/Customization/sockets/Red" .

Do you have any information on how to handle this?

Best regards

**Comment by zlatinz** [comment]
Hi @Gabenny,
You are probably looking for window.history.pushState
See https://stackoverflow.com/questions/824349/how-do-i-modify-the-url-without-reloading-the-page

I would suggest instead of making it as URL path, use the hash.
So it would be **MyDev#Customization/sockets/Red** instead of MyDev/Customization/sockets/Red
